### PR TITLE
zeroc-ice-36: 3.6.3 -> 3.6.5, fix; zeroc-ice: 3.7.2 -> 3.7.6, enable tests; both: clarify license

### PR DESCRIPTION
**Pull request by hrhino**
###### Motivation for this change

- `zeroc-ice-36` is broken
- `zeroc-ice` does not run tests
- The license for both is specified as `gpl2` where `gpl2Only` [is the correct license](https://github.com/zeroc-ice/ice/blob/3.7/README.md#copyright-and-license)
- Both are out of date

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: lib/default.nix [hunk 1]
```diff
@@ -91,7 +91,7 @@ let
       concatImapStringsSep makeSearchPath makeSearchPathOutput
       makeLibraryPath makeBinPath optionalString
       hasInfix hasPrefix hasSuffix stringToCharacters stringAsChars escape
-      escapeShellArg escapeShellArgs replaceChars lowerChars
+      escapeShellArg escapeShellArgs escapeRegex replaceChars lowerChars
```
Comment (hrhino) [inline lib/default.nix:94]: If there was a reason for this not being included, I don't understand what it was.

File: pkgs/development/libraries/zeroc-ice/3.6.nix [hunk 2]
```diff
@@ -53,10 +54,5 @@ stdenv.mkDerivation rec {
     rm -rf $out/share/slice
   '';
 
-  meta = with lib; {
-    homepage = "http://www.zeroc.com/ice.html";
-    description = "The internet communications engine";
-    license = licenses.gpl2;
-    platforms = platforms.unix;
-  };
```
Comment (hrhino) [inline pkgs/development/libraries/zeroc-ice/3.6.nix:66]: also, this one had no maintainer.

File: pkgs/development/libraries/zeroc-ice/uninitialized-variable-warning.patch [hunk 3]
```diff
@@ -0,0 +1,20 @@
+diff --git a/test/Glacier2/dynamicFiltering/TestControllerI.h b/test/Glacier2/dynamicFiltering/TestControllerI.h
```
Comment (hrhino) [inline pkgs/development/libraries/zeroc-ice/uninitialized-variable-warning.patch:1]: My suspicion is that a compiler update sets this warning by default now...

File: pkgs/development/libraries/zeroc-ice/default.nix [hunk 4]
```diff
@@ -52,12 +53,23 @@ in stdenv.mkDerivation rec {
...
   enableParallelBuilding = true;
 
   outputs = [ "out" "bin" "dev" ];
 
+  doCheck = true;
+  checkInputs = with python3.pkgs; [ passlib ];
+  checkPhase = with lib; let
+    # these tests require network access so we need to skip them.
+    brokenTests = map escapeRegex [
+      "Ice/udp" "Glacier2" "IceGrid/simple" "IceStorm" "IceDiscovery/simple"
```
Comment (hrhino) [inline pkgs/development/libraries/zeroc-ice/default.nix:65]: this excludes about 20 of 90 tests, so I think it's still worth it.

File: pkgs/development/libraries/zeroc-ice/3.6.nix [hunk 5]
```diff
@@ -27,13 +27,6 @@ stdenv.mkDerivation rec {
   '';
 
   patches = [
-    # Fixes compilation issues with GCC 8 using one of the patches
-    # provided in https://github.com/zeroc-ice/ice/issues/82
-    ( fetchpatch {
-      url = "https://github.com/zeroc-ice/ice/commit/a6a4981616b669432ff7b588179d6e93694d9e3f.patch";
-      sha256 = "17j5r7gsa3izrm7zln4mrp7l16h532gvmpas0kzglybicbiz7d56";
-      stripLen = 1;
-    })
```
Comment (hrhino) [inline pkgs/development/libraries/zeroc-ice/3.6.nix:36]: This patch was [applied to 3.7](https://github.com/zeroc-ice/ice/commit/a6a4981616b669432ff7b588179d6e93694d9e3f).

**Comment by hrhino** [comment]
Result of `nixpkgs-review pr 135087` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>zeroc-ice-cpp11</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>murmur (murmur_git)</li>
    <li>zeroc-ice (zeroc_ice)</li>
    <li>zeroc-ice-36</li>
  </ul>
</details>

File: pkgs/development/libraries/zeroc-ice/3.6.nix [hunk 6]
```diff
@@ -27,13 +28,8 @@ stdenv.mkDerivation rec {
   '';
 
   patches = [
-    # Fixes compilation issues with GCC 8 using one of the patches
-    # provided in https://github.com/zeroc-ice/ice/issues/82
-    ( fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/zeroc-ice/3.6.nix:32]: Please remove the unused fetchpatch from inputs.

File: pkgs/development/libraries/zeroc-ice/default.nix [hunk 7]
```diff
@@ -51,12 +53,23 @@ in stdenv.mkDerivation rec {
...
+    ];
+  in ''
+    runHook preCheck
+    ${python3}/bin/python3 ./cpp/allTests.py --rfilter='${concatStringsSep "|" brokenTests}'
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/zeroc-ice/default.nix:69]:
```suggestion
    ${python3.interpreter} ./cpp/allTests.py --rfilter='${concatStringsSep "|" brokenTests}'
```

File: pkgs/development/libraries/zeroc-ice/3.6.nix (hunk 6 above)
Comment (hrhino) [reply pkgs/development/libraries/zeroc-ice/3.6.nix:32]: good catch; thanks

File: pkgs/development/libraries/zeroc-ice/default.nix (hunk 7 above)
Comment (hrhino) [reply pkgs/development/libraries/zeroc-ice/default.nix:69]: fancy!

**Comment by hrhino** [comment]
Result of `nixpkgs-review pr 135087` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>murmur (murmur_git)</li>
    <li>zeroc-ice (zeroc_ice)</li>
    <li>zeroc-ice-36</li>
    <li>zeroc-ice-cpp11</li>
  </ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 135087` at b6266dce run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>murmur</li>
    <li>zeroc-ice</li>
    <li>zeroc-ice-36</li>
    <li>zeroc-ice-cpp11</li>
  </ul>
</details>

**Pull request merged**